src/Compilers/CSharp/Test/Semantic/Semantics/SemanticErrorTests.cs

```diff
@@ -8734,9 +8734,9 @@ unsafe public static void Main()
 }
 ";
             CreateStandardCompilation(text, options: TestOptions.UnsafeReleaseDll).VerifyDiagnostics(
-                // (20,23): error CS0254: The right hand side of a fixed statement assignment may not be a cast expression
+                // (20,23): error CS9365: The given expression cannot be used in a fixed statement
```

Comment:
Same here.

Reply:
I think the old message was pretty random here. It only looks like it is more appropriate. The expression that we try to fix makes no sense in many ways. 
I think this is the case where user of `fixed` should read documentation or check web for examples. It is not a lot to ask when _unsafe_ features are involved.



---
In reply to: [167711048](https://github.com/dotnet/roslyn/pull/24469#discussion_r167711048) [](ancestors = 167711048)